dqvm/src/distributed-qvm.lisp

```diff
@@ -0,0 +1,147 @@
+;;; distributed-qvm.lisp
+;;;
+;;; Author: Juan M. Bello-Rivas
+
+(in-package #:dqvm2)
+
+(defclass distributed-qvm (qvm:classical-memory-mixin)
+  ((addresses
+    :accessor addresses
+    :initarg :addresses
+    :initform (error-missing-initform :addresses)
+    :type addresses
+    :documentation "Address table.")
+   (amplitudes
+    :accessor amplitudes
+    :initform nil
+    :type static-vector-pointer        ; qvm::quantum-state
```

Comment:
I would prefer to use the QVM's allocation API if possible, unless there's good justification to avoid it.